### ISSUE TITLE
Web Relay to other devices on a remote device's LAN

### ISSUE THREAD

**Issue by W1BTR** [Ylianst/MeshCentral#6082]
**Is your feature request related to a problem? Please describe.**
#4172 is related, but if I understand correctly, that issue only allows you to relay to a local hosted website.

**Describe the solution you'd like**
Like what is included in Webmin, I'd like to be able to visit websites in the meshcentral browser as if I was running the browser on that machine.

Most of my remote management involves connecting to an agent using meshcentral, controlling the desktop, and accessing other devices on the lan using their web interface.

I'd like to be able to use meshcentral as a relay to access websites or devices on the network via http/https.

**Describe alternatives you've considered**
Webmin has this functionality.

**Additional context**

**Comment by si458** [comment]
yes this is indeed not possible currently but a brilliant idea 💡 
you can specify a different port by right clicking the HTTP/HTTPS links and set the ports,
but the web relay will still only talk to localhost:PORT

**Comment by si458** [comment]
one idea you can try
1. create a `no agent devices relayed thru agent` group and set the agent as a device inside the network
2. add device with IP addresses to that group
3. then click them and use the HTTP/HTTPS buttons
4. (optional right click the HTTP/HTTPS strings to change which port you want to access)

**Comment by PathfinderNetworks** [comment]
This is currently possible.  I use it all the time for exactly the same scenario.  I often need to access the Web GUI of devices on my customers networks (printers, etc). You just use the MeshCentral Router to do it.  
Just launch the MeshCentral Router using the 'Router' link at the bottom of your MeshCentral 'My Devices' page.  
In the Router app click on the Mappings tab.  Click the Add Relay Map button.  Give the mapping a name.  The protocol and Local Port can usually stay as default (unless you have some specific device that uses UDP, etc).  The key is to choose the correct Device Group- as you need to choose a device to relay through that is on the customer network.  Once you have the correct Device Group selected then choose a Relay Device.  I usually try to choose a server or some other device I know is on 24/7.  
The choose the correct application type, for me it's typically HTTPS but some older devices still only use HTTP.  Then, for the Remote IP, you enter the IP address of the device you are wanting to access (such as the printer IP address).  The 'Remote Port' can be changed to whatever is appropriate if that device isn't using the standard port 80, 443, etc that the 'Application' selection uses. 
Once you click OK it will add that new Relay Map to the router.  Then you just click it to launch. 

The only downside to this is that the Relay Mapping you added isn't persistent- meaning it doesn't get saved to the MeshCentral Router app.  Once you close the app that mapping will be lost and you'll have to create it again the next time you want to use it.  It would be very nice if we could save the mappings so they appear at each launch of the MeshCentral Router.

**Comment by ITCMD** [comment]
This is amazing! I never fully understood what the router was for. This is so cool!

**Comment by W1BTR** [comment]
That's exactly what I was looking for. Definitely wish we could save it some way.

Wondering if we should close this request, or perhaps leave it open to suggest more clear documentation and UI guiding, and perhaps the ability to save?

**Comment by si458** [comment]
@W1BTR docs! We need docs! Plz help with docs 🙏 
https://ylianst.github.io/MeshCentral/
All the docs are inside the docs folder in the repo!

**Comment by W1BTR** [comment]
> @W1BTR docs! We need docs! Plz help with docs 🙏 https://ylianst.github.io/MeshCentral/ All the docs are inside the docs folder in the repo!

I had no idea this site existed! The mapping description does a good job explaining it! I will say, Im rather confused on what custom apps does at this point. Ill have to play around some time soon!
